code/modules/surgery/organs/liver.dm

```diff
@@ -29,11 +29,13 @@
 				//handle liver toxin filtration
 				var/toxamount
 				var/static/list/listOfToxinsInThisBitch = typesof(/datum/reagent/toxin)
-				for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
+				for(var/I in listOfToxinsInThisBitch)
+					var/datum/reagent/toxin = I

```

Comment:
This is not a reference to anything, listofToxin.... is a typelist.

Reply:
C.reagents is the toxin you want to reference it seems, then check for each reagent in C.reagents that it's in the list (again right now it's inside out)